src/hotspot/share/opto/parseHelper.cpp

```diff
@@ -680,3 +679,39 @@ void PEAState::validate() const {
   });
 }
 #endif
+
+MergeProcessor::MergeProcessor(PEAState& target) : _state(target) {
+}
+void MergeProcessor::merge(const PEAState& newin, GraphKit* kit, RegionNode* region, int pnum) {
+  // intersection of two aliases
+  Unique_Node_List alias1;
+  Unique_Node_List alias2;
+
+  _state.aliases()->iterate([&](Node* alias, ObjID obj) {
+                             alias1.push(alias);
+                             return true;
+                           });
+  newin.aliases()->iterate([&](Node* alias, ObjID obj) {
+                            alias2.push(alias);
```

Comment:
If we are going to `and` the two alias lists together, can't we just push to `alias1` here?

Reply:
I intend to have the intersection of 2 aliases set. 
alias1 = _state.aliases()->keys();
alias2 = newin.aliases()->key();

```
  VectorSet set = alias1.member_set();
  set &= alias2.member_set();
  alias1.remove_useless_nodes(set);
```